lib/join/joinv1/messages.go

```diff
@@ -199,13 +199,13 @@ func tokenInitFromMessage(msg *messages.TokenInit) (*joinv1.TokenInit, error) {
 
 func clientParamsToMessage(req *joinv1.ClientParams) (messages.ClientParams, error) {
 	var msg messages.ClientParams
-	switch req.GetPayload().(type) {
+	switch payload := req.GetPayload().(type) {
 	case *joinv1.ClientParams_HostParams:
-		msg.HostParams = hostParamsToMessage(req.GetHostParams())
+		msg.HostParams = hostParamsToMessage(payload.HostParams)
 	case *joinv1.ClientParams_BotParams:
-		msg.BotParams = botParamsToMessage(req.GetBotParams())
+		msg.BotParams = botParamsToMessage(payload.BotParams)
 	default:
-		return msg, trace.BadParameter("unrecognized ClientParams payload type %T", req.Payload)
+		return msg, trace.BadParameter("unrecognized ClientParams payload type %T", payload)
```

Comment:
It looks like `responseToMessage`, `resultToMessage` and `challengeToMessage` are also accessing `r.Payload` without checking nil. I think these might also risk panic?

Reply:
yeah i was focusing on the requests handled server-side rather than the responses handled client-side, but these are now updated too